Commands/AnnouncementCmds.cs

```diff
@@ -265,41 +268,50 @@ public async Task AnnounceBuildSlashCommand(SlashCommandContext ctx,
+            [Parameter("role2"), Description("The short name for the second role to ping. Optional.")] string role2Name = null
         )
         {
-            DiscordRole discordRole;
-
-            if (Program.cfgjson.AnnouncementRoles.ContainsKey(roleName))
+            // Validate msg ID
+            DiscordMessage msg;
+            try
             {
-                discordRole = await ctx.Guild.GetRoleAsync(Program.cfgjson.AnnouncementRoles[roleName]);
-                await discordRole.ModifyAsync(mentionable: true);
-                try
-                {
-                    await ctx.Message.DeleteAsync();
-                    var msg = await ctx.Channel.GetMessageAsync(messageId);
-                    await msg.ModifyAsync($"{discordRole.Mention} {content}");
-                }
-                catch
-                {
-                    // We still need to remember to make it unmentionable even if the msg fails.
-                }
-                await discordRole.ModifyAsync(mentionable: false);
+                msg = await ctx.Channel.GetMessageAsync(Convert.ToUInt64(messageId));
             }
-            else
+            catch
             {
-                await ctx.RespondAsync($"{Program.cfgjson.Emoji.Error} That role name isn't recognised!");
+                await ctx.RespondAsync($"{Program.cfgjson.Emoji.Error} That message ID wasn't recognised!", ephemeral: true);
+                return;
+            }
+            
+            if (msg.Author.Id != ctx.Client.CurrentUser.Id)
+            {
+                await ctx.RespondAsync($"{Program.cfgjson.Emoji.Error} That message ID wasn't recognised!", ephemeral: true);
                 return;
             }
+            
+            // Validate roles
+            if (!Program.cfgjson.AnnouncementRoles.ContainsKey(role1Name) || (role2Name is not null && !Program.cfgjson.AnnouncementRoles.ContainsKey(role2Name)))
+            {
+                await ctx.RespondAsync($"{Program.cfgjson.Emoji.Error} The role name(s) you entered aren't recognised!", ephemeral: true);
+                return;
+            }
+            if (role1Name == role2Name)
+            {
+                await ctx.RespondAsync($"{Program.cfgjson.Emoji.Warning} You provided the same role name twice! Did you mean to use two different roles?", ephemeral: true);
+                return;
+            }
+            
+            EditAnnounceCache = (Convert.ToUInt64(messageId), role1Name, role2Name);
+
+            await ctx.RespondWithModalAsync(new DiscordInteractionResponseBuilder().WithTitle("Edit Announcement").WithCustomId("editannounce-modal-callback").AddComponents(new DiscordTextInputComponent("New announcement text. Do not include roles!", "editannounce-modal-new-text", value: Regex.Replace(msg.Content, Constants.RegexConstants.role_rx.ToString(), ""), style: DiscordTextInputStyle.Paragraph)));
```

Comment:
I think that removing and re-adding the role mentions doesn't make sense in the new context since the original text is prefilled verbatim and they could theoretically be anywhere in the text, it may make more sense to keep them and only re-add them if they're not already somewhere within the text.